src/escpos/escpos.py

```diff
@@ -733,6 +738,18 @@ def panel_buttons(self, enable=True):
         else:
             self._raw(PANEL_BUTTON_OFF)
 
+    def queryStatus(self):
+        self._raw(RT_STATUS_ONLINE)
+        time.sleep(1)
+        status = self._read();
+        if (len(status) > 0):
+            return status;
+        else:
+            return [8];
+
+    def isOnline(self):
+        status = self.queryStatus()[0];
+        return ((status & RT_MASK_ONLINE) == 0);
```

Comment:
Well, keep in mind the printer returns some bits fixed to 1. The normal online status is:
00010110 (22)
But the only bit that marks the printer is online is the 4th (bit3): if 0, the printer is online, if 1, offline.

The bool(status is [8]) alternative doesn't look feasible to me, but I could be wrong.
Regarding changing the ==0 with a 'not', yes, it works. I'll push the change in a minute.